src/components/PhoneInput/CountrySelectComponent.jsx

```diff
@@ -0,0 +1,79 @@
+// @flow
+import React, { Component } from 'react';
+import Select from 'react-select';
+
+import FlagComponent from './FlagComponent';
+import OptionComponent from './OptionComponent';
+import ValueComponent from './ValueComponent';
+
+import styles from './style.scss';
+
+export type CountrySelectComponentProps = {|
+  name?: string,
+  value?: ?string,
+  onChange: (value: ?string) => void,
+  onFocus: () => void,
+  onBlur: () => void,
+  options: Array<Country>,
+  disabled?: boolean,
+  tabIndex?: number | string,
+  className?: string,
+  focusPhoneInputField: () => void,
+  name: ?string
+|};
+
+export default class CountrySelectComponent extends Component<CountrySelectComponentProps> {
+  placeholder() {
+    return (
+      <div>
+        <FlagComponent country="none" />
+        <span className={styles.dialCode}>+</span>
+      </div>
+    );
+  }
+
+  handleBlur = (evt: SyntheticFocusEvent<>) => {
+    const {
+      onBlur,
+      focusPhoneInputField,
+    } = this.props;
+
+    onBlur(evt);
+    focusPhoneInputField();
+  };
+
+  handleChange = (item) => {
+    this.props.onChange(item.value);
+  };
+
+  render() {
+    const {
+      className,
+      disabled,
+      name,
+      options,
+      onFocus,
+      value,
+      tabIndex,
+    } = this.props;
+
+    return (
+      <Select
+        className={className}
+        options={options}
+        optionRenderer={OptionComponent}
```

Comment:
Since the `OptionComponent` is only used in this component, it should be within it (just like the placeholder). It would help with readability.

Is there a reason for it to be separate?